File: Arrays/Medium/PrefixSum/PrefixSum.py
# Copyrights to venkys.io
# For more programs visit venkys.io
# Python program for PrefixSum

def prefixSum(arr):
    # Iterate through the array starting from index 1
    for i in range(1, len(arr)):
        # Update each element by adding the value of the preceding element
        arr[i] += arr[i - 1]
    # Print the resulting array after the prefix sum calculation
    print(*arr)

if __name__ == "__main__":
    # Read input from stdin
    input_str = input()

    # Convert the input string to an array (list) of integers
    arr = [int(num) for num in input_str.split()]

    # Call the prefixSum function with the input array
    prefixSum(arr)

File: Arrays/Medium/PrefixSum/PrefixSum.cpp
and the previous element
        arr[i] += arr[i - 1];
    }

    // Print the resulting vector after prefix sum calculation
    for (int num : arr) {
        std::cout << num << " ";
    }

    // Print a new line to separate the output
    std::cout << std::endl;
}

// Main function
int main() {
    // Read the number of elements from the user
    // std::cout << "Enter the number of elements for the array: ";
    int num_elements;
    std::cin >> num_elements;

    // Create a vector named arr with user-input values
    std::vector<int> arr(num_elements);
    // std::cout << "Enter the elements of the array separated by space: ";
    for (int i = 0; i < num_elements; ++i) {
        std::cin >> arr[i];
    }

    // Call the prefixSum function to calculate and print the prefix sum
    prefixSum(arr);

    // Return 0 to indicate successful execution
    return 0;
}
